src/libponyrt/ds/hash.h

```diff
@@ -13,15 +13,33 @@ PONY_EXTERN_C_BEGIN
 #define HASHMAP_BEGIN ((size_t)-1)
 #define HASHMAP_UNKNOWN ((size_t)-1)
 
-/** Definition of a quadratic probing hash map.
+#ifdef PLATFORM_IS_ILP32
+  typedef uint32_t bitmap_t;
+  #define HASHMAP_BITMAP_TYPE_SIZE 32
+#else
+  typedef uint64_t bitmap_t;
+  #define HASHMAP_BITMAP_TYPE_SIZE 64
+#endif
+
+/** Definition of a hash map entry for uintptr_t.
+ */
+typedef struct hashmap_entry_t
+{
+  void* ptr;         /* pointer */
+  size_t hash;       /* hash */
+} hashmap_entry_t;
+
+/** Definition of a linear probing robin hood hash map
+ *  with backward shift deletion.
```

Comment:
Thank you for including the algorithm name for future readers of the code :1st_place_medal: